src/compiler/checker.ts

```diff
@@ -17825,6 +17825,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function addTypeToIntersection(typeSet: Map<string, Type>, includes: TypeFlags, type: Type) {
+        if (type === silentNeverType) {
+            return includes;
+        }

```

Comment:
This makes `silentNeverType` behave more like `unknown` in intersections. Which is definitely not correct for a never type... but a `silentNeverType` is really just a special marker type so it doesn't quite have to obey the rules.